packages/modelserver-client/src/model-server-client-api-v2.ts

```diff
@@ -0,0 +1,133 @@
+/*********************************************************************************
+ * Copyright (c) 2022 STMicroelectronics and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0, or the MIT License which is
+ * available at https://opensource.org/licenses/MIT.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR MIT
+ *********************************************************************************/
+import { Operation } from 'fast-json-patch';
+
+import { ServerConfiguration, SubscriptionOptions } from './model-server-client-api-v1';
+import { Model, ModelServerMessage } from './model-server-message';
+import { ModelServerElement } from './model/base-model';
+import { ModelServerCommand } from './model/command-model';
+import { Diagnostic } from './model/diagnostic';
+import { SubscriptionListener } from './subscription-listener';
+import { AnyObject, TypeGuard } from './utils/type-util';
+
+/** JSON formats supported by the V2 client API. */
+export type JsonFormat = 'json' | 'json-v2';
+
+/** Message formats supported by the V2 client API. */
+export type Format = 'xml' | JsonFormat;
+
+/**
+ * Basic client API to interact with a model server that conforms to the Modelserver API Version 2.
+ */
+export interface ModelServerClientApiV2 {
+    /**
+     * The `initialize` method should be executed before any other requests to the model server.
+     * The given base url should point to the location of the model server API entry point.
+     * (e.g. `http://localhost:8081/api/v1/`). Once the initialization is completed the client is expected
+     * to be ready and should be able to handle REST requests to & responses from the model server.
+     * Any requests to the model server before the client has been initialized are expected to fail (i.e. throw an error)
+     * @param baseUrl Url pointing to the API entry point
+     * @param defaultFormat Optional fallback format that should used when a request method is invoked and no explicit format argument
+     *                      has been passed. The default-default is `'json'`
+     */
+    initialize(baseUrl: string, defaultFormat?: Format): void | Promise<void>;
+
+    /**
+     * Retrieves all available models of the current workspace.
+     */
+    getAll(): Promise<Model[]>;
```

Comment:
It would be great if the V2 API would be documented as well from the beginning. I assume most methods can reuse the documentation from V1.